packages/component/src/Utils/TypeFocusSink/Box.js

```diff
@@ -0,0 +1,93 @@
+import memoize from 'memoize-one';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import Context from './Context';
+import getTabIndex from './getTabIndex';
+import inputtableKey from './inputtableKey';
+
+const DEFAULT_STYLE = { outline: 0 };
+
+export default class Box extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.focus = this.focus.bind(this);
```

Comment:
Focus really is hard to do particularly when you add in scrolling and virtualization (if that is here). Are we messing with the default focus stuff in the browser - mostly does tab still work for navigation? Accessibility can be hard if we want to drive our own tabs.

Reply:
The carousel we are using is actually a native scroll bar, no virtualization (but a good call). Since it is native, it's compatible with browser focus.

In here, `TypeFocusSink` is trying to do one thing: anyone do a `keyDown` inside this `<div>`, we will send the focus to the send box immediately.

We try not to look at every `keyDown`, for example, if the user press `Home` or `Tab` key, we will not focus on the send box. We will only send focus if the user press on one of the "inputtable" key (i.e. A-Z, 0-9) and CTRL-V.